examples/Atlas/runAtlasManipSplit.m

```diff
@@ -0,0 +1,135 @@
+function runAtlasManipSplit(example_options)
+%NOTEST 
+% Run the new split QP controller, which consists of separate PlanEval
+% and InstantaneousQPController objects. The controller will also
+% automatically transition to standing when it reaches the end of its walking
+% plan.
+% @option use_mex [1] whether to use mex. 0: no, 1: yes, 2: compare mex and non-mex
+% @option use_bullet [false] whether to use bullet for collision detect
+% @option quiet [true] whether to silence timing printouts
+
+% 
+% this function is tested in test/testSplitWalking.m
+
+checkDependency('gurobi');
+checkDependency('lcmgl');
+
+if nargin<1, example_options=struct(); end
+example_options = applyDefaults(example_options, struct('use_mex', true,...
+                                                        'use_bullet', false,...
+                                                        'quiet', true,...
+                                                        'terrain', RigidBodyFlatTerrain));
+
+% silence some warnings
+warning('off','Drake:RigidBodyManipulator:UnsupportedContactPoints')
+warning('off','Drake:RigidBodyManipulator:UnsupportedVelocityLimits')
+
+% construct robot model
+options.floating = true;
+options.ignore_self_collisions = true;
+options.ignore_friction = true;
+options.dt = 0.002;
+options.terrain = example_options.terrain;
+options.use_bullet = example_options.use_bullet;
+r = Atlas(fullfile(getDrakePath,'examples','Atlas','urdf','atlas_minimal_contact.urdf'),options);
+r = r.removeCollisionGroupsExcept({'heel','toe'});
+r = compile(r);
+
+% set initial state to fixed point
+load(fullfile(getDrakePath,'examples','Atlas','data','atlas_fp.mat'));
+if isfield(options,'initial_pose'), xstar(1:6) = options.initial_pose; end
+xstar = r.resolveConstraints(xstar);
+r = r.setInitialState(xstar);
+
+v = r.constructVisualizer;
+v.display_dt = 0.01;
+
+nq = getNumPositions(r);
+nv = getNumVelocities(r);
+
+x0 = xstar;
+q0 = x0(1:nq);
+v0 = x0(nq+(1:nv));
+% Plan footsteps to the goal
+pelvis = r.findLinkId('pelvis');
+cnstr = fix_feet_cnstr(r,x0);
+r_hand = r.findLinkId('r_hand');
+r_hand_pt = [0;-0.2;0];
+kinsol0 = r.doKinematics(q0,v0);
+r_hand_pos0 = r.forwardKin(kinsol0,r_hand,r_hand_pt,2);
+r_hand_cnstr = {WorldEulerConstraint(r,r_hand,[nan;pi/2;nan],[nan;pi/2;nan],[0.3 0.7]),...
+  WorldEulerConstraint(r,r_hand,[nan;pi/4;nan],[nan;0.6*pi;nan],[1,1])};
+
+qtraj0 = PPTrajectory(zoh([0,1],[q0 q0]));
+t_plan = [0,0.5,1];
+[xtraj,info] = inverseKinTraj(r,t_plan,qtraj0,qtraj0,cnstr{:},r_hand_cnstr{:});
+
+n_breaks = 20;
+t_breaks = linspace(t_plan(1),t_plan(end)*4,n_breaks);
+r_hand_breaks = zeros(6,n_breaks);
+x_breaks = xtraj.eval(t_breaks);
+qtraj_pp = spline(t_breaks,[zeros(nq,1) x_breaks(1:nq,:) zeros(nq,1)]);
+for i = 1:n_breaks
+  kinsol = r.doKinematics(x_breaks(1:nq,i),x_breaks(nq+(1:nv),i));
+  r_hand_breaks(:,i) = r.forwardKin(kinsol,r_hand,zeros(3,1),1);
+end
```

Comment:
I add the unwrap function, it does not change the behavior. I can confirm that applying the unwrap function does not change r_hand_breaks value. The problem happens at the singularity of pitch = pi/2, that singularity would cause a large value in `qddot`